### PR TITLE
SEAB-6738: Tweak Automatic DOI Section

### PR DESCRIPTION
**Pull request by svonworl**
**Description**
This PR enhances the new "Automatic Dockstore DOI Generation" section.  It is an attempt to codify the bits of information present in this Slack thread: https://ucsc-gi.slack.com/archives/C05EZH3RVNY/p1730923309122769

**Issue**
https://ucsc-cgl.atlassian.net/browse/SEAB-6738

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Ensure that the PR targets the correct branch. Check the milestone or fix version of the ticket.
- [x] If the PR has any new pages, only **AFTER** the PR is approved by all reviewers, generate a new discourse topic, by running the action at https://github.com/dockstore/dockstore-documentation/actions/workflows/add-discourse-topic.yml. Otherwise we may end up with orphaned topics, which would need manual management.

### CODE REVIEW

File: docs/end-user-topics/dois.rst [hunk 1]
```diff
@@ -69,10 +69,17 @@ Automatic Dockstore DOI Generation
...
+
+* The entry is published on Dockstore
+* The version has at least one author in its metadata
+* The tag was created after the entry was published, or the tag is one of the 10 most recent
+
+Automatically-generated DOIs:
+
+* Do not require a Zenodo account
+* Are minted in the Zenodo Dockstore community
+* Will be editable via a linked Zenodo account in a subsequent Dockstore release
```
Comment (denis-yuen) [inline docs/end-user-topics/dois.rst:82]: Or by request to the Dockstore team

File: docs/end-user-topics/dois.rst [hunk 2]
```diff
@@ -69,10 +69,18 @@ Automatic Dockstore DOI Generation
 .. note::
     This feature is currently only available on request. Please contact the Dockstore team at `discuss.dockstore.org <https://discuss.dockstore.org/t/opening-helpdesk-tickets/1506>`__ if you want to participate.
 
-* Automatically mints DOIs for all Git tags pushed to GitHub repositories that have a corresponding published entry in Dockstore
-* The DOIs are minted in the Zenodo Dockstore community
-* No Zenodo account is required to mint DOIs
-* A linked Zenodo account is required to edit DOIs
+Dockstore will automatically mint a DOI for a version that corresponds to a Git tag, if:
+
+* The entry is published on Dockstore
```
Comment (coverbeck) [inline docs/end-user-topics/dois.rst:74]: Do you mean if the entry is already published, or the act of publishing? I'd read it as the former, but then looking at line 76, I think you mean the act of publishing. If so, I suggest something like "Publishing an entry on Dockstore".

File: docs/end-user-topics/dois.rst [hunk 3]
```diff
@@ -69,10 +69,18 @@ Automatic Dockstore DOI Generation
...
+* The version has at least one author in its metadata
+* The tag was created after the entry was published, or the tag is one of the 10 most recent
+
+Automatically-generated DOIs:
+
+* Do not require a Zenodo account
+* Are minted in the Zenodo Dockstore community
+* Are editable by request to the Dockstore team
+* Will be editable via a linked Zenodo account in a subsequent Dockstore release
```
Comment (coverbeck) [inline docs/end-user-topics/dois.rst:83]: @kathy-t Should confirm, but I don't think the Zenodo account will need to be linked to Dockstore. I could be wrong, but I thought this feature will generate a signed Zenodo URL that anybody in possession of can edit with a Zenodo account.

**Review by coverbeck — APPROVED** [review]

File: docs/end-user-topics/dois.rst (hunk 3 above)
Comment (svonworl) [reply docs/end-user-topics/dois.rst]: I removed "linked", we can add it back later if it turns out that it's accurate and important to note...

File: docs/end-user-topics/dois.rst (hunk 2 above)
Comment (svonworl) [reply docs/end-user-topics/dois.rst:74]: > Do you mean if the entry is already published, or the act of publishing? I'd read it as the former, but then looking at line 76, I think you mean the act of publishing. If so, I suggest something like "Publishing an entry on Dockstore".

Both meanings. I changed the last bullet to clarify the two cases a little better.
Might not describe all of the scenarios 100% precisely, but short and readable and more future proof than trying to quantify the retroactive DOI generation that we'll probably do later but don't right now...

File: docs/end-user-topics/dois.rst [hunk 4]
```diff
@@ -69,10 +69,18 @@ Automatic Dockstore DOI Generation
...
     This feature is currently only available on request. Please contact the Dockstore team at `discuss.dockstore.org <https://discuss.dockstore.org/t/opening-helpdesk-tickets/1506>`__ if you want to participate.
 
-* Automatically mints DOIs for all Git tags pushed to GitHub repositories that have a corresponding published entry in Dockstore
-* The DOIs are minted in the Zenodo Dockstore community
-* No Zenodo account is required to mint DOIs
-* A linked Zenodo account is required to edit DOIs
+Dockstore will automatically mint a DOI for a version that corresponds to a Git tag, if:
+
+* The entry is published on Dockstore
+* The version has at least one author in its metadata
```
Comment (kathy-t) [inline docs/end-user-topics/dois.rst:75]: The version also has to be valid https://github.com/dockstore/dockstore/blob/d8e12f97434dc60437e8e6bfffbec5ac00da798b/dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/ZenodoHelper.java#L965

File: docs/end-user-topics/dois.rst (hunk 3 above)
Comment (kathy-t) [reply docs/end-user-topics/dois.rst]: Yeah that's correct, the user won't need to link their account to Dockstore, but they will need to sign into Zenodo after they click the secret Zenodo URL in order to edit the DOI record

File: docs/end-user-topics/dois.rst
Comment (kathy-t) [inline docs/end-user-topics/dois.rst:1]: should we add a reference to this page from https://github.com/dockstore/dockstore-documentation/blob/develop/docs/advanced-topics/snapshot-and-doi.rst so users can see that there are other options?

Comment (svonworl) [reply docs/end-user-topics/dois.rst:1]: > should we add a reference to this page from https://github.com/dockstore/dockstore-documentation/blob/develop/docs/advanced-topics/snapshot-and-doi.rst so users can see that there are other options?

Good idea, but would probably suggest that we do it when the Automatic DOI feature is out of "preview".

**Review by david4096 — APPROVED** [review]

File: docs/end-user-topics/dois.rst [hunk 5]
```diff
@@ -69,10 +69,18 @@ Automatic Dockstore DOI Generation
...
+
+* The entry is published on Dockstore
+* The version is valid and has at least one author in its metadata
+* The tag is created after you publish the entry, or the tag is one of the 10 most recent when you publish the entry
+
+Automatically-generated DOIs:
+
+* Do not require a Zenodo account
+* Are minted in the Zenodo Dockstore community
+* Are editable by request to the Dockstore team
```
Comment (denis-yuen) [inline docs/end-user-topics/dois.rst:82]: To be clear, we probably won't do the editing but can attach their account to the entry in zenodo and let them edit it, sort of a manual version of the next point

**Review by denis-yuen — APPROVED** [review]
Minor clarification, not sure if worth pointing out

**Review by kathy-t — APPROVED** [review]

**Pull request merged**